### PR TITLE
MDNS: cap buffer size for request to min expected length

### PR DESCRIPTION
**Pull request by sandeepmistry**
As discussed in https://github.com/arduino-libraries/WiFi101/issues/155#issuecomment-300911833.

cc/ @washcroft

### CODE REVIEW

File: src/WiFiMDNSResponder.cpp [hunk 1]
```diff
@@ -135,9 +135,14 @@ bool WiFiMDNSResponder::parseRequest()
...
-    // read packet
-    uint8_t request[packetLength];
-    udpSocket.read(request, packetLength);
+    // read up to the min expect request length
+    uint8_t request[minimumExpectedRequestLength];
+    udpSocket.read(request, minimumExpectedRequestLength);
+
+    // discard the rest
+    while(udpSocket.available()) {
+      udpSocket.read();
```
Comment (washcroft) [inline src/WiFiMDNSResponder.cpp]: Thanks, unfortunately I think there is something deeper here... the MKR1000 still crashes when handling these large UDP packet streams, the crash occurs inside the while loop when discarding the rest of the data.

**Comment by sandeepmistry** [comment]
Hi @washcroft,

Would you mind trying out my latest change in this pull request? There was a memory corruption issue when a UDP packet with 1438 or more bytes was received.

**Comment by washcroft** [comment]
Hello @sandeepmistry,
Thank you, I was able to find the source of the large packets on my network and use that to reproduce the crashing. However, with your latest change, I am no longer able to reproduce the crashing with these large packets.

**Comment by sandeepmistry** [comment]
@washcroft great, thanks for trying it out! These changes will be included in the next library release.

**Pull request merged**